analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -30,6 +31,10 @@ operation switch
             {
                 _ when IParameterReferenceOperationWrapper.IsInstance(operation) => IParameterReferenceOperationWrapper.FromOperation(operation).Parameter,
                 _ when ILocalReferenceOperationWrapper.IsInstance(operation) => ILocalReferenceOperationWrapper.FromOperation(operation).Local,
+                _ when IFieldReferenceOperationWrapper.IsInstance(operation)
+                    && IFieldReferenceOperationWrapper.FromOperation(operation) is var fieldReference
+                    && (fieldReference.Instance == null // static fields
+                        || fieldReference.Instance.IsAnyKind(OperationKindEx.InstanceReference)) => fieldReference.Field,
```

Comment:
_Optional:_ I find this a bit too complicated for a switch expression. But I'm not very convinced about that.

Maybe this can improve it, but feel free not to do it:
```suggestion
                    && IsStaticOrThis(fieldReference.Instance) => fieldReference.Field,
```